nemo/collections/asr/data/audio_to_label.py

```diff
@@ -304,26 +318,37 @@ def __init__(
         max_duration: Optional[float] = None,
         trim: bool = False,
         load_audio: bool = True,
+        is_regression_task: bool = False,
     ):
         super().__init__()
         self.collection = collections.ASRSpeechLabel(
-            manifests_files=manifest_filepath.split(','), min_duration=min_duration, max_duration=max_duration,
+            manifests_files=manifest_filepath.split(','),
+            min_duration=min_duration,
+            max_duration=max_duration,
+            is_regression_task=is_regression_task,
         )
 
         self.featurizer = featurizer
         self.trim = trim
         self.load_audio = load_audio
+        self.is_regression_task = is_regression_task
 
-        self.labels = labels if labels else self.collection.uniq_labels
-        self.num_classes = len(self.labels)
+        if not self.is_regression_task:
 
-        self.label2id, self.id2label = {}, {}
-        for label_id, label in enumerate(self.labels):
-            self.label2id[label] = label_id
-            self.id2label[label_id] = label
+            self.labels = labels if labels else self.collection.uniq_labels
 
-        for idx in range(len(self.labels[:5])):
-            logging.debug(" label id {} and its mapped label {}".format(idx, self.id2label[idx]))
+            self.num_classes = len(self.labels)
+
+            self.label2id, self.id2label = {}, {}
+            for label_id, label in enumerate(self.labels):
+                self.label2id[label] = label_id
+                self.id2label[label_id] = label
+
+            for idx in range(len(self.labels[:5])):
+                logging.debug(" label id {} and its mapped label {}".format(idx, self.id2label[idx]))
+
+        else:
+            self.num_classes = len(self.labels) if self.labels is not None else 1
```

Comment:
self.labels is not set before this else clause, its set inside the if clause. So this will raise an error since self doesnt contain the attribute `labels`.